kernel/kernel-api/src/main/java/io/delta/kernel/types/StructField.java

```diff
@@ -198,6 +209,19 @@ public StructField withDataType(DataType newType) {
     return new StructField(name, newType, nullable, metadata, typeChanges);
   }
 
+  public static StructField createInternalColumn(StructField field) {
```

Comment:
Can we put this somewhere else so that it's not part of the public interface of `StructField`? We really don't want engines to create internal columns

Reply:
I moved the method to `ScanImpl`